src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,272 @@
+    #
+    #     xy_col_counts: Dict[str, int] = {}
+    #     for col in list(new_df.columns):
+    #         if any([col.endswith(x) for x in ['_x', '_y']]):
+    #             common_col = col[:-2]
+    #             if common_col not in xy_col_counts:
+    #                 xy_col_counts[common_col] = 0
+    #             xy_col_counts[common_col] += 1
+    #
+    #     xy_cols: List[str] = [k for k, v in xy_col_counts.items() if v == 2]
+    #     for col in xy_cols:
+    #         new_df[col] = new_df[col + '_y'].fillna(new_df[col + '_x'])
+    #         new_df = new_df.drop([col + '_x', col + '_y'], axis=1)
+    #
+    #     return new_df
+
+    # df['term_info_source'] = ''
+    # results_df_excluded_kids['term_info_source'] = 'icd10cm.ttl'
+    # results_df_included_kids['term_info_source'] = 'icd10cm.ttl'
+    # # Determine which factors were not in source
+    term_in_icd10cm_ttl = set(xxx['term_id'])
+    df['term_in_source_icd10cm.ttl'] = df['term_id'].apply(
+        lambda term_id: term_id in term_in_icd10cm_ttl)
+
+    # Join
+    # df2 = pd.merge(df, results_df_excluded_kids, how='left', on='term_id')
+    df3 = pd.merge(df, xxx, how='left', on='term_id').fillna('')
+    print(list(df3.columns)) # TODO temp
+
+    # Restructure
+    # todo: repurpose
+    df4 = df3
+    # df4['PK_unique_path'] = \
+    #     df4.apply(lambda row: row['term_range_id'].replace('ICD10CM:', '') + '_' + row['term_id'].replace(
+    #         'ICD10CM:', '') + '_' + row['child_id'].replace('ICD10CM:', ''), axis=1)
+    # df4['PK_unique_path'] = df4['PK_unique_path'].apply(lambda x: x[:len(x) - 1] if x.endswith('_') else x)
+    # df4 = df4.rename(columns={
+    #     'exclusion_reason': 'child_exclude_reason',
+    #     'exclude_children': 'child_exclude'})
+    # df4 = df4[[
+    #     'PK_unique_path', 'term_range_id', 'term_range_label', 'term_id', 'term_label', 'child_id',
+    #     'child_label', 'term_info_source', 'term_in_source_icd10cm.ttl', 'child_exclude', 'child_exclude_reason']]
+    # df4 = df4.sort_values(['PK_unique_path', 'term_range_id', 'term_id', 'child_id'])
+
+    # TODO 2. regex -----------------------------------------------------------
+    #  Lastly the table should support the following trick: If a value in term_label starts with REGEX:, the script
+    #  should search for all labels in the ontology and exclude them. For example:
+    #  example table:
+    #  term_id	term_label	exclusion_reason	exclude_children
+    #  | REGEX:^.*phenotype$ | phenotype | yes
+    #  Should match all labels then end in the word phenotype and then exclude all their children.
+    #  First here, I should filter original df to have only regex rows
+    pass
+
+    # TODO: JOIN non-regex and regex results
+    pass
+
+    # todo: save
+    #  2. a simple two column file with the term_id and the exclusion reason
+    #    - Fields: `Term ID`, `Exclusion Reason` || `term_id`, `exclusion_reason`
```

Comment:
@matentzn You wrote in the GH issue to name these columns `Term ID` and `Exclusion Reason`, but I'm wondering if you'd rather leave them as `term_id` and `exclusion_reason` for consistency / ease of downstream processing?

Reply:
I think keep as `term_id` for downstream processing related reasons.. Thank you!